### PR TITLE
JU-280:upgrade java version and all other dependencies

### PR DESCRIPTION
**Pull request by dgurjar-ch**
Upgraded java to java 21
Removed old Health check and added Spring boot actuator
Updated all the dependency to make them work with java 21

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,28 +15,46 @@
...
-        <spring-boot-maven-plugin.version>2.5.5</spring-boot-maven-plugin.version>
+        <java.version>21</java.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven-compiler-plugin.version>3.12.1</maven-compiler-plugin.version>
+        <maven-surefire-plugin.version>3.2.5</maven-surefire-plugin.version>
+
+        <org.mapstruct.version>1.5.5.Final</org.mapstruct.version>
+        <oracle.jdbc11.version>23.4.0.24.05</oracle.jdbc11.version>
+
+        <spring-boot-dependencies.version>3.2.4</spring-boot-dependencies.version>
```
Comment (hwatkins-ch) [inline pom.xml:26]: 3.2.5 is the latest version of spring

File: pom.xml [hunk 2]
```diff
@@ -15,28 +15,46 @@
...
-        <structured-logging.version>1.9.12</structured-logging.version>
-        <environment-reader-library.version>1.3.6</environment-reader-library.version>
-        <system-rules-version>1.17.2</system-rules-version>
+
+        <environment-reader-library.version>3.0.1</environment-reader-library.version>
+
+        <system-rules-version>1.19.0</system-rules-version>
+
+
         <hamcrest-all-version>1.3</hamcrest-all-version>
```
Comment (hwatkins-ch) [inline pom.xml:36]: Can we upgrade this to new hamcrest dependency and use version 2.2, hamcrest-all is depreciated.

https://mvnrepository.com/artifact/org.hamcrest/hamcrest/2.2

File: src/test/java/uk/gov/companieshouse/chd/order/api/service/OrderServiceTest.java [hunk 3]
```diff
@@ -2,18 +2,14 @@
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.Is.is;
-import static org.junit.Assert.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/chd/order/api/service/OrderServiceTest.java:5]: We have been avoiding wildcard imports :) Can we replace with implicit imports

**Review by hwatkins-ch — COMMENTED** [review]
Looks good, few small comments

File: pom.xml [hunk 4]
```diff
@@ -102,22 +129,36 @@
...
             <groupId>com.github.stefanbirkner</groupId>
             <artifactId>system-rules</artifactId>
             <version>${system-rules-version}</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>uk.org.webcompere</groupId>
+            <artifactId>system-stubs-core</artifactId>
+            <version>2.1.6</version>
```
Comment (foman-ch) [inline pom.xml:142]: could this version be moved to the properties above

File: pom.xml [hunk 5]
```diff
@@ -102,22 +129,36 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>uk.org.webcompere</groupId>
+            <artifactId>system-stubs-jupiter</artifactId>
+            <version>2.1.6</version>
```
Comment (foman-ch) [inline pom.xml:148]: same with this?

File: src/test/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesControllerIntegrationTest.java [hunk 6]
```diff
@@ -38,29 +38,12 @@ class MissingImageDeliveriesControllerIntegrationTest {
...
-    }
-
     @Test
     @DisplayName("Create Missing Image Delivery returns 201 on valid headers")
     void createMissingImageDeliverySucceedsOnValidHeaders() throws Exception {
-
-        final String content = objectMapper.writeValueAsString(MISSING_IMAGE_DELIVERIES_DTO);
+        MissingImageDeliveriesDTO missingImageDeliveriesDTO1 = missingImageDeliveriesDTO();
+        missingImageDeliveriesDTO1.setCompanyName("Test");
+         String content = objectMapper.writeValueAsString(missingImageDeliveriesDTO1);
```
Comment (foman-ch) [inline src/test/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesControllerIntegrationTest.java:46]: indentation looks off

**Review by foman-ch — DISMISSED** [review]

**Comment by hwatkins-ch** [comment]
Change merge branch, we don't want to merge into master

**Review by hwatkins-ch — CHANGES_REQUESTED** [review]
Change target merge branch to a new migration branch

File: pom.xml (hunk 2 above)
Comment (dgurjar-ch) [reply pom.xml:36]: Actually in virtual repository latest version still 1.3 
http://repository.aws.chdev.org:8081/artifactory/virtual-release/org/hamcrest/hamcrest-all/

File: src/test/java/uk/gov/companieshouse/chd/order/api/service/OrderServiceTest.java (hunk 3 above)
Comment (dgurjar-ch) [reply src/test/java/uk/gov/companieshouse/chd/order/api/service/OrderServiceTest.java:5]: Done, imported individual package

File: pom.xml (hunk 4 above)
Comment (dgurjar-ch) [reply pom.xml:142]: Done

File: pom.xml (hunk 5 above)
Comment (dgurjar-ch) [reply pom.xml:148]: Done

File: pom.xml (hunk 1 above)
Comment (dgurjar-ch) [reply pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (dgurjar-ch) [reply pom.xml]: Done

**Pull request merged**